plugins/nv-filters/nvidia-audiofx-filter.c

```diff
@@ -520,36 +565,60 @@ static void nvidia_audio_update(void *data, obs_data_t *s)
 				if (err != NVAFX_STATUS_SUCCESS) {
 					do_log(LOG_ERROR, "NvAFX_SetFloat(Intensity Ratio: %f) failed, error %i",
 					       ng->intensity_ratio, err);
-					nvidia_audio_destroy(ng);
+					nvidia_audio_disable(ng);
 				}
 			}
 			pthread_mutex_unlock(&ng->nvafx_mutex);
 		}
+		/* updating for VAD toggled on or off */
+		if (nvafx_new_sdk) {
+			if (vad != ng->vad && (strcmp(ng->fx, method) == 0)) {
+				NvAFX_Status err;
+				ng->vad = vad;
+				pthread_mutex_lock(&ng->nvafx_mutex);
+				for (size_t i = 0; i < ng->channels; i++) {
+					err = NvAFX_SetU32(ng->handle[i], NVAFX_PARAM_ENABLE_VAD, ng->vad);
+					if (err != NVAFX_STATUS_SUCCESS) {
+						do_log(LOG_ERROR, "NvAFX_SetU32(VAD: %i) failed, error %i", ng->vad,
+						       err);
+						nvidia_audio_disable(ng);
+					}
+				}
+				pthread_mutex_unlock(&ng->nvafx_mutex);
+			}
+		}
 		/* swapping to a new FX requires a reinitialization */
 		if ((strcmp(ng->fx, method) != 0)) {
 			pthread_mutex_lock(&ng->nvafx_mutex);
 			bfree((void *)ng->fx);
 			ng->fx = bstrdup(method);
 			ng->intensity_ratio = intensity;
+			ng->vad = vad;
 			set_nv_model(ng, method);
 			os_atomic_set_bool(&ng->reinit_done, false);
 			for (int i = 0; i < (int)ng->channels; i++) {
 				/* Destroy previous FX */
 				if (NvAFX_DestroyEffect(ng->handle[i]) != NVAFX_STATUS_SUCCESS) {
 					do_log(LOG_ERROR, "FX failed to be destroyed.");
-					nvidia_audio_destroy(ng);
+					nvidia_audio_disable(ng);
 				} else {
 					ng->handle[i] = NULL;
 				}
 			}
 			if (!nvidia_audio_initialize_internal(data))
-				nvidia_audio_destroy(ng);
+				nvidia_audio_disable(ng); //nvidia_audio_destroy(ng);
```

Comment:
Can probably remove the commented portion of this line.

Reply:
done